python/simple/v1/ampli/ampli.py

```diff
@@ -0,0 +1,818 @@
+                self.client = options.client.instance or Amplitude(api_key=api_key, configuration=configuration)
+                self.client.configuration.api_key = api_key
+            else:
+                self.client = Amplitude(api_key=API_KEY[env], configuration=DEFAULT_CONFIGURATION)
+    
+    def initialized_and_enabled(self) -> bool:
+        if not self.client:
+            logging.getLogger(__name__).error("Ampli is not yet initialized. Called `ampli.load()` on app start.")
+            return False
+        return not self.disabled
+    
+    def track(self, user_id: str, event: BaseEvent, event_options: Optional[EventOptions] = None):
+        if not self.initialized_and_enabled():
+            return
+        if not event_options:
+            event_options = EventOptions()
+        if user_id:
+            event_options["user_id"] = user_id
+        event.load_event_options(event_options)
+        self.client.track(event)
+        
+    def identify(self, user_id: str,
+                 identify_properties: IdentifyProperties,
+                 event_options: Optional[EventOptions] = None):
+        if not self.initialized_and_enabled():
+            return
+        if not event_options:
+            event_options = EventOptions()
+        if user_id:
+            event_options["user_id"] = user_id
+        identify_obj = Identify()
+        if identify_properties:
+            for k, v in identify_properties.properties.items():
+                identify_obj.set(k, v)
+        self.client.identify(identify_obj, event_options)
+
+    def group_identify(self, group_type: str,
+                       group_name: str,
+                       group_properties: GroupProperties,
+                       event_options: Optional[EventOptions] = None):
+        if not self.initialized_and_enabled():
+            return
+        identify_obj = Identify()
+        if group_properties:
+            for k, v in group_properties.properties.items():
+                identify_obj.set(k, v)
+        self.client.group_identify(group_type, group_name, identify_obj, event_options)
+
+    def set_group(self, user_id: str,
+                  group_type: str,
+                  group_name: Union[str, List[str]],
+                  event_options: Optional[EventOptions] = None):
+        if not self.initialized_and_enabled():
+            return
+        if not event_options:
+            event_options = EventOptions()
+        if user_id:
+            event_options["user_id"] = user_id
+        self.client.set_group(group_type, group_name, event_options)
+        
+    def flush(self):
+        if not self.initialized_and_enabled():
+            return
+        self.client.flush()
+        
+    def shutdown(self):
+        if not self.initialized_and_enabled():
+            return
+        self.client.shutdown()
+        self.disabled = True
+        
+    def event_max_int_for_test(
+        self,
+        user_id: str,
```

Comment:
To allow setting userId by Event/EventOptions we should allow `user_id: str | None` (still required but can be None)